### PR TITLE
Fix #161138

### LINKED ISSUE #161138: The port flag does not work when starting the server
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: From sources
- OS Version:

Steps to Reproduce:

- Start `server-main` with `--port` flag

### PR DESCRIPTION
**Pull request by mustard-mh**
Fixes microsoft/vscode#161138

### CODE REVIEW

File: src/server-main.js [hunk 1]
```diff
@@ -188,12 +188,11 @@ function sanitizeStringArg(val) {
...
  */
 async function parsePort(host, strPort) {
-	let specificPort;
 	if (strPort) {
 		let range;
 		if (strPort.match(/^\d+$/)) {
-			specificPort = parseInt(strPort, 10);
-			if (specificPort === 0) {
+			const specificPort = parseInt(strPort, 10);
+			if (specificPort) {
```
Comment (mustard-mh) [inline src/server-main.js:195]: I [see](https://github.com/microsoft/vscode/issues/161112) people use `--port 0` , is value zero allowed?

Comment (jeanp413) [reply src/server-main.js:195]: yes `0` is for random port

**Comment by jeanp413** [comment]
cc @aeschli

**Comment by aeschli** [comment]
Thanks a lot! Dup of https://github.com/microsoft/vscode/pull/161254

**Pull request closed without merging**